lib/sources/user/local/local_user_source_impl.dart

```diff
@@ -155,7 +155,14 @@ class LocalUserSourceImpl extends LocalUserSource {
 
       // Store the updated accounts list
       await store.record(ACCOUNTS_LIST_KEY).put(txn, updatedJsonAccounts);
+
+      // Update the current active user if needed
+      final MooncakeAccount currentUser = await getActiveAccount();
+      if (currentUser?.address == data.address) {
+        await _saveActiveAccount(txn, data);
+      }
```

Comment:
Wasn't this previously removed to force us to use the `setActiveAccount` method instead? I think that way is better since this method could be called also when wanting to refresh the non-active accounts to keep the updated. What do you think?

Reply:
I personally think it's fine where it is since the statement won't affect non active users. Otherwise I can do this on the bloc level.

Reply:
@ryuash The problem with this approach is that if we leave it here, then we would have to: 

- update the documentation of the source method
- rename the method to `saveAccountAndSetActive` 
- create another account just for saving an account locally

What if in the future we want to implement a method that allows the refresh of all the accounts (even non active ones) in the background? What method would we use to save those account? 

Since there is already another method that sets an account as active, I also see these lines as redundant to be honest. 

I think it might be better to move it to the BLoC level as you suggested

Reply:
@ryuash Sorry for the mind change but I've actually taken a closer look at the code :see_no_evil: I think that you can roll back to the first implementation without any problem as you were right. Sorry but I was distracted and did not see the `if` condition properly. I thought you were always updating the active account if it existed previously. Really sorry :pray: